test/functional/JLM_Tests/src/org/openj9/test/management/TestOperatingSystemMXBean.java

```diff
@@ -64,6 +65,48 @@
 	private static final int NTHREADS = 1;
 	private static final int NITERATIONS = 10;
 
+	private enum ContainerStatus {
+		UNINITIALIZED,
+		NOT_FOUND,
+		FOUND
+	}
+	private static ContainerStatus inContainer = ContainerStatus.UNINITIALIZED;
+
+	private static boolean isPhysicalMemoryTestExcluded() {
+		/* Exclude when running in a docker container
+		 * https://github.com/eclipse/openj9/issues/12038
+		 */
+		if (inContainer == ContainerStatus.NOT_FOUND) {
+			return false;
+		} else if (inContainer == ContainerStatus.FOUND) {
+			return true;
+		}
+		String osname = System.getProperty("os.name");
+		if ("Linux".equals(osname)) {
+			try (FileInputStream env = new FileInputStream("/.dockerenv")) {
+				logger.info("/.dockerenv found");
+				inContainer = ContainerStatus.FOUND;
+				return true;
+			} catch (IOException e) {
+				// no "/.dockerenv" file, fall through
+			}
+			try (BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/1/cgroup")))) {
+				String line;
+				while ((line = reader.readLine()) != null) {
+					if (line.contains("docker")) {
```

Comment:
Suggested simplification:
```
		if (reader.lines().anyMatch(line -> line.contains("docker"))) {
			logger.info("Container found: " + line);
			inContainer = ContainerStatus.FOUND;
			return true;
		}
```

Reply:
It doesn't work because `line` is used afterwards to log how the container was found, for easier debugging.

Reply:
I missed that aspect. It could probably be reworked, but perhaps it's not worth it.

Reply:
Updated it to use your pattern.

Reply:
The lambda need not change if you use `findFirst()`:
```
	Optional<String> docker = reader.lines().filter(line -> line.contains("docker")).findFirst();
	if (docker.isPresent()) {
		logger.info("Container found: " + docker.get());
		inContainer = ContainerStatus.FOUND;
		return true;
	}
```

Reply:
Updated.